Update dd agent

mbbx6spp opened the pull request:
###### Motivation for this change

`dd-agent` is currently broken on master and 17.03. This happened when this PR was merged, this broke `dd-agent`'s dependency on `docker` python package in NixOS (it depends on 1.10.x, and 2.x changes symbols around): https://github.com/NixOS/nixpkgs/pull/22201/files

I also added missing python package dependencies in `dd-agent` for `python-etcd` and `consul`.

As a separate commit I also updated the `dd-agent` package sources from 5.5.2 -> 5.11.2 and create appropriate git logs as per contributing guide for each.

###### Things done

- [X] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] Linux
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Output from `nox-review`:

```bash
<redacted>
Result in /run/user/1000/nox-review-0i73b82r
total 0
lrwxrwxrwx 1 spotter users 67 Mar  9 20:42 result -> /nix/store/1374i67agihx6n5c1374awvsr08l77rf-python2.7-docker-1.10.6
lrwxrwxrwx 1 spotter users 59 Mar  9 20:42 result-2 -> /nix/store/m2kbqm0l4xqy3bzvngacvraaj6l6978g-dd-agent-5.11.2
lrwxrwxrwx 1 spotter users 67 Mar  9 20:42 result-3 -> /nix/store/wpmiy1wckhmj12zg0m8idg01z5pkpwkf-python3.5-docker-1.10.6
```

FRidh commented:
Thanks. Could you put the `docker_1_10.nix` expression in `pkgs/tools/networking/dd-agent/` and import it straight into the `dd-agent` expression? As long as there's no other applications that need that specific version I don't think we should make it public.

mbbx6spp committed:
dd-agent: fix Python missing dependencies and versions

mbbx6spp committed:
dd-agent: 5.5.2 -> 5.11.2

mbbx6spp commented:
Thanks for the speedy review. @FRidh I kept the two commits, both should still work as expected. I removed the older `pythonPackages.docker_1_10` package from being public and embedded in the package lambda's let. Let me know if there are any other changes.

As a last sanity check I did the following:

1. I removed `docker_1_10` from `buildInputs` list in the `dd-agent` package lambda.

```bash
$ dd-agent -h
Traceback (most recent call last):
  File "/nix/store/g4xqhraqc3q78yp2a639v6j9yrbjmpwh-dd-agent-5.11.2/bin/.dd-agent-wrapped", line 13, in <module>
    from config import get_version, initialize_logging  # noqa
  File "/nix/store/g4xqhraqc3q78yp2a639v6j9yrbjmpwh-dd-agent-5.11.2/agent/config.py", line 29, in <module>
    from utils.service_discovery.config import extract_agent_config
  File "/nix/store/g4xqhraqc3q78yp2a639v6j9yrbjmpwh-dd-agent-5.11.2/agent/utils/service_discovery/config.py", line 9, in <module>
    from utils.service_discovery.sd_backend import SD_BACKENDS
  File "/nix/store/g4xqhraqc3q78yp2a639v6j9yrbjmpwh-dd-agent-5.11.2/agent/utils/service_discovery/sd_backend.py", line 9, in <module>
    from utils.service_discovery.sd_docker_backend import SDDockerBackend
  File "/nix/store/g4xqhraqc3q78yp2a639v6j9yrbjmpwh-dd-agent-5.11.2/agent/utils/service_discovery/sd_docker_backend.py", line 10, in <module>
    from docker.errors import NullResource, NotFound
ImportError: No module named docker.errors
?=1 | Error
```

2. Added it back:

```
$ vim pkgs/tools/networking/dd-agent/default.nix
?=0 | OK

$ nix-env -f . -iA dd-agent
replacing old ‘dd-agent-5.11.2’
installing ‘dd-agent-5.11.2’
?=0 | OK

$ dd-agent -h
Please supply a configuration file at /etc/dd-agent/datadog.conf or in the directory where the Agent is currently deployed.
?=3 | Error
```

FRidh commented:
Thank you. I've pushed this to master. I made some minor changes in https://github.com/NixOS/nixpkgs/commit/35f9f63a39f24f8b7e68451d5860a74260b4e5f8. While in Nix we typically pass in individual packages, in the case of Python we pass the whole package set in order to prevent any accidental mixing.

mbbx6spp commented:
Any chance it can get backported to 17.03?

FRidh commented:
Done!
